### ISSUE TITLE
NRT doesn't check possible null dereference in LINQ

### ISSUE THREAD

**Issue by Pzixel** [dotnet/roslyn#55467]
**Version Used**:  5.0.302

**Steps to Reproduce**:

Consider following code:

```cs
using System;
using System.Collections.Generic;
using System.Linq;

#nullable enable

public class C {
    public void M(IEnumerable<string?> s) {
        var x = from c in s
        select c.Length;
    }
    
    public void N(IEnumerable<string?> s) {
        var x = s.Select(c => c.Length);
    }
}
```

**Expected Behavior**:

Both lines show  "warning CS8602: Dereference of a possibly null reference."

**Actual Behavior**:

LINQ version doesn't produce any warnings and allows to silently defererence null

**Comment by jaredpar** [comment]
Looks like a duplicate of #37468

**Comment by canton7** [comment]
Is that the same? This one is specifically about query syntax not working where the equivalent extension method syntax is fine.

**Comment by Pzixel** [comment]
How is it duplicate? The linked issue is about false positive but this one about not having warning where it should be

**Comment by jaredpar** [comment]
They're duplicates because that is an uber issue for working out the issues with LINQ and NRT. There are several patterns in LINQ that don't work well with NRT analysis. That issue is the uber issue for tracking them as a group

**Comment by canton7** [comment]
@Pzixel perhaps add this description as a comment to the other? I don't see anything in the other about query syntax failing where the equivalent extension method syntax works.

**Comment by Pzixel** [comment]
Could you please add this comment? I'm sorry you lost me, I don't quite get what you want from me :)

**Comment by canton7** [comment]
Copy yourdescription from this issue, at the top of the page, and paste it into a comment on the duplicate issue?

**Comment by Pzixel** [comment]
But it's already there as "mentioned in NRT doesn't check possible null dereference in LINQ", right?

**Comment by canton7** [comment]
People don't tend to click through to closed linked issues -- the assumption is that they just say something that's already been said. But, your problem, you do what you want :)
